### PR TITLE
Storable: fix dependencies and use of note()

### LINKED ISSUE #17422: Storable 3.15 distribution on cpan has broken META.json
... due to this problem it cannot be installed on Perl 5.6.2

https://metacpan.org/source/XSAWYERX/Storable-3.15/META.json

Problems:

1) URL for reporting bugs does not work

2) META.json says that minimal ExtUtils::MakeMaker is 0. But Makefile.PL has `use ExtUtils::MakeMaker 6.31;`. Which cause configure error on 5.6.2:
```
ExtUtils::MakeMaker version 6.31 required--this is only version 6.17 at Makefile.PL line 11.
BEGIN failed--compilation aborted at Makefile.PL line 11.
```
cpan client does not try to install new ExtUtils::MakeMaker version because META.json says that version 6.17 (>= 0) is enough.

3) Test `t/recurse.t` uses `note` function from `Test::More`. This function was introduced in Test::More 0.81 and therefore for running Storable tests is needed at least Test::More in version 0.81. But META.json does not even specify that `Test::More` is needed for tests, nor its minimal required versions. Therefore on 5.6.2 it tails on error:
```
1..39
String found where operator expected at t/recurse.t line 350, near "note 'trying catching recursive aref stack overflow'"
        (Do you need to predeclare note?)
String found where operator expected at t/recurse.t line 365, near "note 'trying catching recursive href stack overflow'"
        (Do you need to predeclare note?)
syntax error at t/recurse.t line 350, near "note 'trying catching recursive aref stack overflow'"
syntax error at t/recurse.t line 365, near "note 'trying catching recursive href stack overflow'"
Execution of t/recurse.t aborted due to compilation errors.
```
As 5.6.2 has bundled only version 0.47. and cpan client did not tried to upgrade it as Storable's META.json does not mention that Test::More is needed.

Due to these problems, as META.json is invalid, it is not possible to Storable on Perl 5.6.2.

And due to problem 1) I do not know where to report bugs for Storable, so I hope that core perl is the proper place.

Btw, new Test::More depends on Storable, so I suggest to remove usage of `note` function (or at least make it optional) as otherwise you create a cyclic dependency between Storable and Test::More which disallow installing any of them on older Perl versions. So adding dependency on Test::More 0.81 is not really a good idea.

### PR DESCRIPTION
**Pull request by tonycoz**
Fixes #17422

### CODE REVIEW

File: dist/Storable/Makefile.PL [hunk 1]
```diff
@@ -16,9 +16,13 @@ WriteMakefile(
...
     DISTNAME            => "Storable",
-# We now ship this in t/
-#    PREREQ_PM           => { 'Test::More' => '0.41' },
-    PREREQ_PM           => { XSLoader => 0 },
+    PREREQ_PM           => { 'Test::More' => '0.41' },
+    PREREQ_PM           =>
+      {
+          XSLoader => 0,
+          'Test::More' => '0.41',
+          'ExtUtils::MakeMaker' => '6.31',
```
Comment (skaji) [inline dist/Storable/Makefile.PL]: If we want to require EUMM before executing Makefile.PL,
we should put it in prereqs/configure/requires section in META files.

File: dist/Storable/Makefile.PL [hunk 2]
```diff
@@ -16,9 +16,13 @@ WriteMakefile(
     AUTHOR              => 'Perl 5 Porters',
     LICENSE             => 'perl',
     DISTNAME            => "Storable",
-# We now ship this in t/
-#    PREREQ_PM           => { 'Test::More' => '0.41' },
-    PREREQ_PM           => { XSLoader => 0 },
+    PREREQ_PM           => { 'Test::More' => '0.41' },
```
Comment (skaji) [inline dist/Storable/Makefile.PL]: Should we remove this line?

Comment (tonycoz) [reply dist/Storable/Makefile.PL]: Done.

File: dist/Storable/Makefile.PL (hunk 1 above)
Comment (tonycoz) [reply dist/Storable/Makefile.PL]: I've updated Makefile.PL to set configure, build, and test requirements.

**Comment by pali** [comment]
Could you please also release a new version on cpan? As the last released version is 3.15 from April 2019: https://metacpan.org/pod/Storable

**Pull request merged**